### PR TITLE
Revamped CMakeLists.txt

### PR DESCRIPTION
**Pull request by GamePad64**
Bumped CMake version to 3.1
Removed WITH_CPP** options, in favor of -DCMAKE_CXX_STANDARD=11
Added public includes to both dynamic and static targets, enabling CMake-idiomatic way to link these libraries with link_library and without include_directory
Fixed "#pragma mark" warnings on GCC

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,8 @@
-cmake_minimum_required(VERSION 2.6.4)
+cmake_minimum_required(VERSION 3.1)
 
 option(WITH_TESTS           "Build tests."       OFF)
 option(WITH_EXAMPLE         "Build example."     OFF)
 option(WITH_STATIC          "Build static libs." ON)
-option(WITH_CPP11           "Build with C++11."  ON)
-option(WITH_CPP14           "Build with C++14."  OFF)
```
Comment (jaredgrubb) [inline CMakeLists.txt]: Why were these removed? These broke the Travis setup.

Comment (GamePad64) [inline CMakeLists.txt]: I see, Travis has only CMake 2.8.
CMake 3.1 supports setting C++ version using CMAKE_CXX_STANDARD variable. It is better, because you can set this variable in parent project, and then include "docopt" directory in it (and link it using link_library()).

I will rewrite travis.yml to update CMake version (to version 3.2.2 for example, as used in Ubuntu Wily) before building the code. Do not close this pull request, please.

Comment (jaredgrubb) [inline CMakeLists.txt]: Oh ok that makes sense. I'm still new to both cmake and Travis, so any help you offer is greatly appreciated!

**Comment by GamePad64** [comment]
Well, I have a little question: Why is there an option WITH_CPP11 in the first place? The library depends on C++11 and it will not build with WITH_CPP11=OFF.

**Comment by GamePad64** [comment]
Okay, it is ready to merge.

**Comment by jaredgrubb** [comment]
You're right about not needing the C++11 option. I cant recall why I thought it was needed!

Thanks for your help on getting this all put together!
